core/src/main/scala/com/databricks/labs/remorph/discovery/SnowflakeTableDefinition.scala

```diff
@@ -0,0 +1,122 @@
+package com.databricks.labs.remorph.discovery
+
+import com.databricks.labs.remorph.parsers.intermediate.{DataType, StructField}
+import com.databricks.labs.remorph.parsers.snowflake.{SnowflakeLexer, SnowflakeParser}
+import org.antlr.v4.runtime.{CharStreams, CommonTokenStream}
+import com.databricks.labs.remorph.parsers.snowflake.SnowflakeTypeBuilder
+
+import java.sql.Connection
+import scala.collection.mutable.ListBuffer
+
+class SnowflakeTableDefinition(conn: Connection) {
+
+  /**
+   * Parses a data type string and returns the corresponding DataType object.
+   *
+   * @param dataTypeString The string representation of the data type.
+   * @return The DataType object corresponding to the input string.
+   */
+  private def getDataType(dataTypeString: String): DataType = {
+    val inputString = CharStreams.fromString(dataTypeString)
+    val lexer = new SnowflakeLexer(inputString)
+    val tokenStream = new CommonTokenStream(lexer)
+    val parser = new SnowflakeParser(tokenStream)
+    val ctx = parser.dataType()
+    val dataTypeBuilder = new SnowflakeTypeBuilder
+    dataTypeBuilder.buildDataType(ctx)
+  }
+
+  private def getTableDefinitionQuery(catalogueName: String): String = {
+    s"""SELECT
+                   |    sft.TABLE_CATALOG,
+                   |    sft.TABLE_SCHEMA,
+                   |    sft.TABLE_NAME,
+                   |    sfe.location,
+                   |    sfe.file_format_name,
+                   |    sfv.view_definition,
+                   |    t.Schema as derivedSchema,
+                   |    sft.BYTES,
+                   |FROM (
+                   |    SELECT
```

Comment:
use CTE (`WITH x AS (SELECT ...`) instead of subqueries.

Reply:
done